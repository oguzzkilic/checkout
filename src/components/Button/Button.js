import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

export default function Button({ isLoading, children, ...rest }) {
  return (
    <StyledButton
      disabled={!!isLoading}
      {...rest}
      status={isLoading ? 'disabled' : rest.status}
    >
      {isLoading && '...'}
      {children}
    </StyledButton>
  );
}

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  isLoading: PropTypes.bool,
};

Button.defaultProps = {
  onClick: f => f,
  isLoading: false,
};

const buttonCss = css`
  font-weight: normal;
  font-family: 'Source Sans Pro', Helvetica, Arial, sans-serif;
  border: none;
  cursor: pointer;
  outline: none;
  background: none;
  border: 1px solid transparent;
  line-height: 1;
  display: block;
  align-items: center;
  vertical-align: middle;
  justify-content: center;
  white-space: nowrap;
  user-select: none;
  font-size: 16px;
  border-radius: 100px;
  width: 100%;
  text-align: center;

  ${props => {
    if (props.status) {
      if (props.status === 'default') {
        return `
          color: #FFF;
          background-color: #1d54ba;
          border: 0;
          padding: 10px 15px;
          width: ${props.fullWidth ? '100%' : ''};
        `;
      }

      if (props.status === 'disabled') {
        return `
          color: #FFF;
          background-color: #7ce09f;
          border: 0;
          padding: 10px 15px;
          width: ${props.fullWidth ? '100%' : ''};
      `;
      }

      if (props.status === 'noBg') {
        return `
          color: #64acff;
          background-color: transparent;
          border: 0;
          font-size: 14px;
        `;
      }

      if (props.status === 'noBg-disabled') {
        return `
          color: #64acff;
          background-color: transparent;
          border: 0;
        `;
      }
    }
  }};
`;

const StyledButton = styled.button`${buttonCss};`;
