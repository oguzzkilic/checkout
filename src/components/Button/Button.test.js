import React from 'react';
import Renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import Button from './Button';

import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

describe('Button Component Tests', () => {
  it('renders correctly', () => {
    const tree = Renderer
      .create(<Button>Click me!</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle the click event', () => {
    const fn = jest.fn(() => true);
    const wrapper = shallow(
      <Button onClick={fn}>Click me!</Button>,
    );
    wrapper.simulate('click');
    expect(fn).toBeCalled();
  });
});
