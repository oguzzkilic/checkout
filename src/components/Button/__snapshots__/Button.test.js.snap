// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Tests renders correctly 1`] = `
.c0 {
  font-weight: normal;
  font-family: 'Source Sans Pro',Helvetica,Arial,sans-serif;
  border: none;
  cursor: pointer;
  outline: none;
  background: none;
  border: 1px solid transparent;
  line-height: 1;
  display: block;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  white-space: nowrap;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-size: 16px;
  border-radius: 100px;
  width: 100%;
  text-align: center;
}

<button
  className="c0"
  disabled={false}
  onClick={[Function]}
>
  Click me!
</button>
`;
